### PR TITLE
johnshew/login

### PR DESCRIPTION
**Pull request by johnshew**
Adds a nodejs server integrated Microsoft graph to store VoTT profile information

### CODE REVIEW

**Comment by johnshew** [comment]
Starting the PR process.  Any suggestions or recommendations appreciated.

File: server/src/config.ts [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+  // Required, must be 'code', 'code id_token', 'id_token code' or 'id_token'
+  // If you want to get access_token, you must use 'code', 'code id_token' or 'id_token code'
+  responseType: 'code id_token',
+
+  // Required
+  responseMode: 'form_post',
+
+  // Required, the reply URL registered in AAD for your app
+
```
[the remark this answers is not in the document]
Comment (johnshew) [reply server/src/config.ts]: Done

File: package-lock.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
     "name": "vott",
-    "version": "2.0.0",
+    "version": "2.1.0",
```
[the remark this answers is not in the document]
Comment (johnshew) [reply package-lock.json]: Not sure why that occurred.  I did not do this.

File: server/public/test.html [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+        }
+        
+        async function displayResponse(response) {
+            let json = null;
+            try { json = await response.json().catch(); } catch(err) { }
+            let result = (response.ok ? '*success*' : '*failed*') + '\n' + (json ? JSON.stringify(json, undefined, 2) : '') ;
+            document.getElementById("result").innerHTML = result;
+        }
+
+        async function test1(e) {
```
[the remark this answers is not in the document]
Comment (johnshew) [reply server/public/test.html]: Done

**Comment by MichaelCVelez** [comment]
Hello @johnshew

I am anxious to try out the login server you have created since this is a functionality I’ve been hoping for in VoTT for quite some time now. Do you have any guidance for running it? I have compiled the development branch and tried executing the server using npm start, but noticed the lib folder is missing since it’s ignored by git.

Do you have steps for how I can run the login feature? 

Thank you in advance

**Pull request merged**